com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+            float dy = rhsKey.value - lhsKey.value;
+            float length = 1.0f / dx;
+            float lengthSqr = length * length;
+
+            float m1 = lhsKey.outTangent;
+            float m2 = rhsKey.inTangent;
+            float d1 = m1 * dx;
+            float d2 = m2 * dx;
+
+            // Note: The coeffecients are calculated to match what the editor does internally. These coeffeceients expect a
+            // t in the range of [0,dx]. We could change the function to accept a range between [0,1], but then this logic would
+            // be different from internal editor logic which could cause subtle bugs later.
 
-        // TODO: Curve interpolation
+            float c0 = (d1 + d2 - dy - dy) * lengthSqr * length;
+            float c1 = (dy + dy + dy - d1 - d1 - d2) * lengthSqr;
+            float c2 = m1;
+            float c3 = lhsKey.value;
+
+            float t = Mathf.Clamp(currTime - lhsKey.time, 0.0f, dx);
+
+            dstValue = (t * (t * (t * c0 + c1) + c2)) + c3;
+            dstDeriv = (t * (3.0f * t * c0 + 2.0f * c1)) + c2;
+        }
+
+        // lhsIndex and rhsIndex are the indices in the keys array. The lhsIndex/rhsIndex may be -1, in which it creates a synthetic first key
+        // at startTime, or beyond the length of the array, in which case it creates a synthetic key at endTime.
+        static private Keyframe EvalKeyAtTime(Keyframe[] keys, int lhsIndex, int rhsIndex, float startTime, float endTime, float currTime)
+        {
+            Assert.IsTrue(keys.Length >= 1);
+            Assert.IsTrue(startTime <= keys[0].time);
+            Assert.IsTrue(keys[keys.Length - 1].time <= endTime);
+
+            Keyframe lhsKey = FetchKeyFromIndexClamped(keys, lhsIndex, startTime, endTime);
+            Keyframe rhsKey = FetchKeyFromIndexClamped(keys, rhsIndex, startTime, endTime);
+
+            float currValue = 0.0f;
+            float currDeriv = 0.0f;
+            EvalCurveSegmentAndDeriv(ref currValue, ref currDeriv, lhsKey, rhsKey, currTime);
+
+            Keyframe dstKey = new Keyframe(currTime, currValue, currDeriv, currDeriv);
+            return dstKey;
+        }
+
+        public override void Interp(AnimationCurve lhsCurve, AnimationCurve rhsCurve, float t)
+        {
+            if (t <= 0.0f)
+            {
+                m_Value = lhsCurve;
+            }
+            else if (t >= 1.0f)
+            {
+                m_Value = rhsCurve;
+            }
+            else
+            {
+                m_Value = new AnimationCurve();
+
+                Assert.IsTrue(lhsCurve.keys.Length >= 1);
+                Assert.IsTrue(rhsCurve.keys.Length >= 1);
+
+                // first, figure out the start and end time to include both curves
+                float startTime = Mathf.Min(lhsCurve.keys[0].time, rhsCurve.keys[0].time);
+                float endTime = Mathf.Max(lhsCurve.keys[lhsCurve.length - 1].time, rhsCurve.keys[rhsCurve.length - 1].time);
```

Comment:
Each time you use `lhsCurve.keys`, a copy is made. Use a variable to store it instead. Same for every array returned by any method that is not a ref return.